[MM-38683] Android Initialisation Bug + BottomSheet Correction

shazm opened the pull request:
#### Summary

Fixes the following:
- First login on Android was crashing the app
- Android keyboard was in front of bottom sheet when called for a Failed Message
- iOS Failed Message bottom sheet was causing the app to crash

Test: Have a message fail to post and then attempt to retry or delete it.

#### Ticket Link
Fixes https://mattermost.atlassian.net/browse/MM-38677

#### Checklist
<!--
Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.
-->


#### Device Information
This PR was tested on: <!-- Device name(s), OS version(s) -->

#### Screenshots
<!--
If the PR includes UI changes, include screenshots/GIFs/Videos (for both iOS and Android if possible).
-->

#### Release Note
```release-note
NONE
```

enahum commented on app/screens/options_modal/options_modal_list.js:
```diff
@@ -87,6 +88,20 @@ export default class OptionsModalList extends PureComponent {
             );
         });
 
+        const cancel = (
```
What is this? On Android it wasn't showing the delete option, how does this solve it?

shazm replied:
Part of cleaning this up I removed the "Cancel" button but didn't pick up on that there were a few places in the codebase that were sending a cancel button as well, but with the index for the cancel button which actually removed it on the assumption that the modal would add one at the bottom. 

I'm not entirely sure where / how this came about; but just reverting it to the state it was before.

enahum commented:
Let's ask QA to test this, then if is in fact working correctly I'll approve

mattermod commented:
Building app in separate branch.

mattermod commented:
Successfully building:  
https://circleci.com/gh/mattermost/mattermost-mobile/18063  
https://circleci.com/gh/mattermost/mattermost-mobile/18062

mattermod commented:
Artifact links:  
https://18063-70265724-gh.circle-artifacts.com/0/tmp/artifacts/Mattermost_Beta.ipa  
https://18062-70265724-gh.circle-artifacts.com/0/tmp/artifacts/Mattermost_Beta.apk

mattermod commented:
Failed to delete ref. @mattermost/core-build-engineers have been notified. Error 
```GET https://api.github.com/repos/mattermost/mattermost-mobile/commits/refs/heads/build-pr-5685-72b1381/status: 502  []```

josephbaylon reviewed (approved):
LGTM. Verified fixes work in this PR.

I'll let @furqanmlk do the final QA.

amyblais commented:
@furqanmlk Please help prioritize this PR review if you can.

furqanmlk commented:
> @furqanmlk Please help prioritize this PR review if you can.

Working right now, will update shortly

furqanmlk reviewed (commented):
Thanks @shazm 
iOS and Android Login crash has been verified.
BottomSheet still hide `Cancel` text

### Before PR in Android 

![image](https://user-images.githubusercontent.com/6909890/134588427-2322a8a0-dc66-4b4b-b14f-e0438ce0d4db.png)


### In Current PR for Android

Login crash has been fixed. ✅ 

But I can not see `Cancel` button in BottomSheet

![image](https://user-images.githubusercontent.com/6909890/134588682-38a7d089-fa53-4ba0-a773-6052f671fb95.png)


### Before PR in iOS

https://user-images.githubusercontent.com/6909890/134588898-cf5a1e98-bf96-4648-816a-dbb522302f77.mp4

### In current PR for iOS

`Delete`,  `retry` and `cancel` working fine

https://user-images.githubusercontent.com/6909890/134589125-f60a3a46-c974-41ba-ab8b-7b3aa11cabe8.mp4

enahum commented:
@shazm will you be making the required changes soon? we need to merge this ASAP, please prioritize

mkraft commented:
@matthewbirtch Just wondering if this bottom drawer should have a "Cancel" option given that other drawers omit an explict cancel option and just let the user click away to cancel.

matthewbirtch commented:
> @matthewbirtch Just wondering if this bottom drawer should have a "Cancel" option given that other drawers omit an explict cancel option and just let the user click away to cancel.

@mkraft I agree. The Cancel option isn't needed here as you can simply display the bottom drawer by tapping away from it. The word cancel is also a little confusion in this context since users might think `Cancel` would stop trying to send the message and delete it. I think we should remove the cancel option

matthewbirtch reviewed (changes requested):
@shazm one thing I noticed:

It appears that extra space is being added under the failed message in the channel. This seems to be happened after I tap on the `!` icon to open the bottom sheet. This appears to only be an issue on iOS

<img src="https://user-images.githubusercontent.com/2040554/134685383-19a1c59b-03be-4440-ab19-9f68a7664b20.PNG" width="414">

enahum commented:
@matthewbirtch can you confirm https://github.com/mattermost/mattermost-mobile/pull/5685#pullrequestreview-763107241 is caused by the recent changes and not happening on 1.46?

matthewbirtch commented:
> @matthewbirtch can you confirm [#5685 (review)](https://github.com/mattermost/mattermost-mobile/pull/5685#pullrequestreview-763107241) is caused by the recent changes and not happening on 1.46?

My apologies. This _is_ indeed happening in 1.46. I'll file a separate bug ticket for this

mkraft commented:
> > @matthewbirtch Just wondering if this bottom drawer should have a "Cancel" option given that other drawers omit an explict cancel option and just let the user click away to cancel.
> 
> @mkraft I agree. The Cancel option isn't needed here as you can simply dismiss the bottom drawer by tapping away from it. The word cancel is also a little confusing in this context since users might think `Cancel` would stop trying to send the message and delete it. I think we should remove the cancel option

Thanks. I removed the cancel button and pushed the commit.

mkraft commented:
@matthewbirtch Here's a screenshot from the iOS simulator without the Cancel button:

![Simulator Screen Shot - iPhone 12 - 2021-09-24 at 11 44 50](https://user-images.githubusercontent.com/1149597/134703580-13d48d1a-1c19-4c27-ba39-26e835dd6fa3.png)

matthewbirtch commented:
Is it easy enough to add icons to the two menu options @mkraft? Most of our bottom sheet menus aim to have icons for quick recognition. If so, I suggest we use the following

Try Again:
![image](https://user-images.githubusercontent.com/2040554/134703997-9158b77e-3523-451b-aba5-1cd091d37633.png)

Delete Message:
![image](https://user-images.githubusercontent.com/2040554/134703950-e40db76b-5f07-404b-9638-62254dbad33d.png)

Let me know what you think

enahum reviewed (changes requested):
iOS is in fact showing both options in the bottom sheet but android is not

matthewbirtch reviewed (approved):
Approving this. Whether you add the icons in this PR or not is fine with me, so I don't want to block.

Just make sure the options are visible on android as per the [comment](https://github.com/mattermost/mattermost-mobile/pull/5685#pullrequestreview-762534055) from @furqanmlk and [comment](https://github.com/mattermost/mattermost-mobile/pull/5685#pullrequestreview-763235268) from @enahum

enahum commented:
Try changing the style in `options_modal_list` to

```
optionContainer: {
        alignSelf: 'stretch',
        backgroundColor: 'white',
        borderTopLeftRadius: 12,
        borderTopRightRadius: 12,
        paddingVertical: 10,
        ...Platform.select({
            ios: {
                paddingBottom: 25,
            },
            android: {
                marginBottom: 25,
            },
        }),
    },
```

furqanmlk commented:
@mkraft 

iOS is working fine

But in Android now, `Delete` text getting hide 
![image](https://user-images.githubusercontent.com/6909890/134715518-8cac31d2-8bd6-4fc3-b788-8584028937a3.png)

shazm commented:
@furqanmlk could you share your device / os versions? I'm testing on a Pixel 4 and it seems fine:
<img width="395" alt="Screen Shot 2021-09-27 at 11 06 46 am" src="https://user-images.githubusercontent.com/456511/134831390-d5907f09-bfe9-479f-bc8d-7ace434ad128.png">

enahum commented:
@shazm I got the same result as @furqanmlk on a Pixel 4

shazm commented:
I think there might be something happening for physical vs on-screen system buttons for Android depending on the API level.

I added a `marginBottom: -10` which aligns that bottom sheet appropriately on a Pixel 4 (API 29). It looks a bit low on a Pixel 4a (physical device), so I removed it and adding a `paddingBottom: 10`; which now looks like this:

Pixel 4 (API 29) Simulator:

<img width="404" alt="Screen Shot 2021-09-27 at 5 19 55 pm" src="https://user-images.githubusercontent.com/456511/134862701-e3b8e316-4f66-4cc3-92a2-b91c64a013aa.png">

Pixel 4a (API30 / Physical Device):
![Screenshot (27 Sep 2021 5_20_15 pm)](https://user-images.githubusercontent.com/456511/134862804-ce775c92-78fe-48b6-a09a-6bd4cbf0b7ea.png)

There's also an option to always show the "Back / Home / App Switch" buttons on the bottom of the device which I enabled; getting:

![Screenshot (27 Sep 2021 5_25_24 pm)](https://user-images.githubusercontent.com/456511/134863183-627c4ba1-4f64-4775-93f1-7cd856a38274.png)

Pixel 4XL (API30):
<img width="411" alt="Screen Shot 2021-09-27 at 5 28 17 pm" src="https://user-images.githubusercontent.com/456511/134863466-97ed5000-4b93-43b1-afab-7070159b9e1f.png">

I think the API versions render differently rather than the device itself; will continue testing across API28/29/30.

shazm commented:
Rebased with latest master, and includes fix for API 28/29 for the bottom margin; otherwise test across API 27/28/29/30 on a Pixel 4 and bottom sheet looks like it should with no clipping.

enahum reviewed (approved):
After the changes I made to the styling this now works fine for me on Android & iOS without having to set specifics for Android SDK versions.

mattermod commented:
Building app in separate branch.

mattermod commented:
Successfully building:  
https://circleci.com/gh/mattermost/mattermost-mobile/18115  
https://circleci.com/gh/mattermost/mattermost-mobile/18114

furqanmlk reviewed (approved):
Thanks @shazm 

I am using Pixel 3a. 
All looks good now

![image](https://user-images.githubusercontent.com/6909890/134917902-1acdcbad-1227-4db1-b87a-2f94f7bb5ac7.png)

mattermod commented:
Artifact links:  
https://18115-70265724-gh.circle-artifacts.com/0/tmp/artifacts/Mattermost_Beta.ipa  
https://18114-70265724-gh.circle-artifacts.com/0/tmp/artifacts/Mattermost_Beta.apk

mattermod commented:
Cherry pick is scheduled.
